Tensile/KernelWriterAssembly.py

```diff
@@ -7409,6 +7473,140 @@ def shiftVectorComponents(self, kernel, tP):
     self.vgprPool.checkIn(vReg)
     return kStr
 
+  def shiftVectorComponentsForMatrixInst(self, kernel, tP):
+    kStr = ""
+
+    # glvw
+    glvw = tP["glvw"]
+
+    numThreadInWave            = 64
+    numContinuousOutput        = 4
+    numOutputThreads1          = kernel["MatrixInstN"]
+    numOutputThreads0          = numThreadInWave // numOutputThreads1
+    numSubOutputPerWave0       = numOutputThreads0 * numContinuousOutput
+    numSubOutputGroupsPerWave0 = kernel[tP["mt"]] // numSubOutputPerWave0
+    numShiftBlock              = numContinuousOutput // glvw
+    numOutputElements          = numSubOutputGroupsPerWave0 * numContinuousOutput
+
+    subTile1 = (kernel["MacroTile1"] // ((kernel["SubGroup0"] * kernel["SubGroup1"]) // numThreadInWave)) // kernel["MatrixInstN"]
+
+    # labels
+    svrLabels = []
+    sviLabels = []
+    svoLabels = []
+    for i in range(0, glvw):
+      r = (i+1) % glvw
+      label = self.getLabelNum("ShiftVectorComponents%u_R%u" % (tP["idx"], r) )
+      svrLabels.append(label)
+      tmpLabels = []
+      tmp2Labels = []
+      for v in range(0, numSubOutputGroupsPerWave0):
+        label = self.getLabelNum("ShiftVectorComponents%u_R%u_V%u" % (tP["idx"], r, v) )
+        tmpLabels.append(label)
+        tmp2Labels2 = []
+        for o in range(0, numShiftBlock):
+          label = self.getLabelNum("ShiftVectorComponents%u_R%u_V%u_O%u" % (tP["idx"], r, v, o) )
+          tmp2Labels2.append(label)
+        tmp2Labels.append(tmp2Labels2)
```

Comment:
Is this a typo?

Reply:
no, we have two level label.

Reply:
update comment and make a slide to explain in e-mail